src/items.rs

```diff
@@ -25,6 +27,54 @@ use syntax::print::pprust;
 use syntax::parse::token;
 
 impl<'a> FmtVisitor<'a> {
+    pub fn visit_let(&mut self, local: &ast::Local, span: Span) {
+        self.format_missing_with_indent(span.lo);
+
+        // New scope so we drop the borrow of self (context) in time to mutably
+        // borrow self to mutate its buffer.
+        let result = {
+            let context = self.get_context();
+            let mut result = "let ".to_owned();
+
+            // 1 = ;
+            // FIXME(#82): reserving space for the semicolon overly conservative
+            // for multi-line expressions.
```

Comment:
I think we have different interpretation of the meaning of the width parameter of Rewrite.
If we reformat a block of code like this
```
|..........+---------------+|
|    +-----|               ||
|    |                     ||
|    |                     ||
|    |                     ||
|    |                  +--+|
|    +------------------+...|
```
(taken from a comment of @nrc in #133))
I understand `width` as a limit only for the last line, while this comment assumes it's a limit for all lines.
We should clarify this in the contract of Rewrite (eventually, I think we will need the two different width limits : one for last line, and one for all other lines).